server/src/main/java/org/candlepin/pinsetter/tasks/UndoImportsJob.java

```diff
@@ -129,9 +133,26 @@ public void toExecute(JobExecutionContext context) throws JobExecutionException
 
             log.info("Deleting all pools originating from manifests for owner/org: {}", displayName);
 
+            // BZ 1452694: Don't delete pools for custom subscriptions
+            // We need to verify that we aren't deleting pools that are created via the API.
+            // Unfortunately, we don't have a 100% reliable way of detecting such pools at this point,
+            // so we'll do the next best thing: In standalone, pools with an upstream pool ID are those
+            // we've received from an import (and, thus, are eligible for deletion). In hosted,
+            // however, we *are* the upstream source, so everything is eligible for removal.
+            // This is pretty hacky, so the way we go about doing this check should eventually be
+            // replaced with something more generic and reliable, and not dependent on the config.
+
+            // Note: this logic is duplicated in CandlepinPoolManager.refreshPoolsWithRegeneration
+
+            // TODO:
+            // Remove the standalone config check and replace it with a check for whether or not the
+            // pool is "managed"  -- however we decide to implement that in the future.
```

Comment:
While such a change would be optimal, I feel it would be a rather large change, and one I'm not sure we want to do with this PR. It's something that's better done with the pool hierarchy task.